src/ducks/translations.js

```diff
@@ -0,0 +1,96 @@
+import apiClient from 'panoptes-client/lib/api-client';
+
+// Actions
+const FETCH_TRANSLATIONS_ERROR = 'FETCH_TRANSLATIONS_ERROR';
+const FETCH_TRANSLATIONS = 'FETCH_TRANSLATIONS';
+const FETCH_TRANSLATIONS_SUCCESS = 'FETCH_TRANSLATIONS_SUCCESS';
+const SET_STRINGS = 'SET_STRINGS';
+
+const TRANSLATION_STATUS = {
+  IDLE: 'translation_status_idle',
+  FETCHING: 'translation_status_fetching',
+  READY: 'translation_status_ready',
+  ERROR: 'translation_status_error'
+};
+
+const initialState = {
+  status: TRANSLATION_STATUS.IDLE,
+  strings: {
+    field_guide: {},
+    tutorial: {}
+  }
+};
+
+const translationsReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case SET_STRINGS:
+      const strings = Object.assign({}, state.strings, action.payload);
+      return Object.assign({}, state, { strings });
+
+    case FETCH_TRANSLATIONS:
+      return Object.assign({}, state, {
+        status: TRANSLATION_STATUS.FETCHING
+      });
+
+    case FETCH_TRANSLATIONS_SUCCESS:
+      return Object.assign({}, state, {
+        status: TRANSLATION_STATUS.READY
+      });
+
+    case FETCH_TRANSLATIONS_ERROR:
+      return Object.assign({}, state, {
+        status: TRANSLATION_STATUS.ERROR
+      });
+
+    default:
+      return state;
+  }
+};
+
+const loadTranslations = (translated_type, translated_id) => {
+  return (dispatch, getState) => {
+    const languages = ['ar', 'he', 'en'];
+    dispatch({
+      type: FETCH_TRANSLATIONS,
+      translated_type,
+      translated_id
+    });
+    languages.map((language) => {
+      apiClient.type('translations')
```

Comment:
When we call an asynchronous `dispatch(fetchWhatever())` that uses `apiClient`, we want to make sure it returns a Promise. i.e. we want to make sure we write `return apiClient.type('whatever')...`

In this case, I'm not sure what loadTranslations is returning, if it's returning anything. `undefined` maybe?

The reason I need to check this is because a few lines below this, we have...
```
const loadResources = () => {
  return (dispatch) => {
    Promise.all([
      dispatch(loadTranslations('field_guide', '40')),
      dispatch(loadTranslations('tutorial', '207'))
    ]).then(() => {
      dispatch({ type: FETCH_TRANSLATIONS_SUCCESS });
    }).catch(() => {
      dispatch({ type: FETCH_TRANSLATIONS_ERROR });
    });
  };
};
```
...and IF `dispatch(loadTranslations())` returns `undefined`, then `Promise.all` will cause `FETCH_TRANSLATIONS_SUCCESS` to trigger BEFORE all the `loadTranslation()` calls succeed.

I'm going to test this and see whattup.

Reply:
BOO! Promises are a bit of a headache, as usual. I modded to code to look like this...

```
const loadTranslations = (translated_type, translated_id) => {
  return (dispatch, getState) => {
    //bla bla bla
    languages.map((language) => {
      apiClient.type('translations')
        .get({ translated_type, translated_id, language })
        .then(([translations]) => {
          if (translations) {
            //bla bla bla
            console.log('+++ GOT IT FOR ', language);
            dispatch({
              type: SET_STRINGS,
              payload: {
                [translated_type]: resource
              }
            });
          }
        });
    });
  };
};

const loadResources = () => {
  return (dispatch) => {
    Promise.all([
      dispatch(loadTranslations('field_guide', '40')),
      dispatch(loadTranslations('tutorial', '207'))
    ]).then(() => {
      console.log('+++ SUCCESS');
      dispatch({ type: FETCH_TRANSLATIONS_SUCCESS });
    }).catch(() => {
      console.log('+++ FAILURE');
      dispatch({ type: FETCH_TRANSLATIONS_ERROR });
    });
  };
};
```

...and sure enough, console.log looks like this:
```
+++ SUCCESS
+++ GOT IT FOR  en
+++ GOT IT FOR  he
+++ GOT IT FOR  ar
+++ GOT IT FOR  ar
+++ GOT IT FOR  he
+++ GOT IT FOR  en
```

OK, bad news and good news. 

Bad news, the current way that this PR does fetch translations doesn't work, because `FETCH_TRANSLATIONS_SUCCESS` triggers _before_ all translations are fully fetched.

Good news... this _may_ not matter, if we've been REALLY robust with our existential checks for `reduxStore.translations.strings` - i.e. we check that `reduxStore.translations.strings.['english']` exists before we render `reduxStore.translations.strings.['english'].burpingSounds`

Reply:
Ok, this makes sense. I recall seeing some questionable behaviors in writing this code initially that supports your assumptions above. This is something I'll have to nail out in a future commit based on your review of this PR.